src/hooks/useSwapStats.ts

```diff
@@ -0,0 +1,89 @@
+import { BLOCK_TIME, ChainId } from "../constants"
+import { useQuery } from "@tanstack/react-query"
+
+const swapStatsURI = "https://ipfs.saddle.exchange/swap-stats.json"
+
+type PoolStats = {
+  oneDayVolume: string
+  apy: string
+  tvl: string
+  utilization: string
+}
+type SwapStats = {
+  [chainId in ChainId]?: Partial<{
+    [swapAddress: string]: PoolStats
+  }>
+}
+
+type SwapStatsReponse = {
+  [chainId in ChainId]?: {
+    [swapAddress: string]: {
+      oneDayVolume: number
+      APY: number
+      TVL: number
+    }
+  }
+}
+export function useSwapStats(): {
+  swapStats: SwapStats | undefined
+  swapStatsLoading: boolean
+} {
+  const {
+    data: swapStats,
+    isLoading: swapStatsLoading,
+    isSuccess,
+  } = useQuery<SwapStatsReponse>(
+    ["swapStats"],
+    async () => {
+      const res = await fetch(`${swapStatsURI}`)
+      if (res.status >= 200 && res.status < 300) {
+        return res.json()
+      }
+      throw new Error("Unable to fetch swap stats from IPFS")
+    },
+    { retry: 3, refetchInterval: BLOCK_TIME * 3 },
+  )
+  if (isSuccess) {
```

Comment:
instead of manually handling undefined you should be able to use `selector` as described here: https://tkdodo.eu/blog/react-query-data-transformations#3-using-the-select-option

this only runs if query isn't undefined

Reply:
I didn't know about the `select`. Will update!

Reply:
I've never used it myself so it's up to you

Reply:
I love this way.